# Exploratory Data Analysis

## Importing libraries

In [65]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt  
import seaborn as sns

## Reading the dataset 

In [46]:
df = pd.read_csv('../data/raw/AB_NYC_2019.csv')
df.sample(10, random_state = 1)

id                                               name    host_id  \
18907  15019959                       Amazing huge furnished room!     445894   
46663  35373956          Fun 3 Bedroom Just Minutes to Manhattan!!  266299609   
19757  15818624          Sunny and spacious  1-bedroom in Brooklyn   18192430   
9705    7460933     Large Private Room in Greenpoint/ Williamsburg   12738054   
3322    1983531   Spacious Bright 1 Bedroom East Village Apartment    5586949   
12637   9705782                  Spacious Brooklyn Brownstone 3BR+    8144242   
19383  15468765       Quiet & Spacious Brooklyn Apt for Cat-Lovers   12063606   
35358  28047011  Huge room in great area, 25 minutes from Manha...   70260514   
28001  21884557               Brooklyn Home in the Heart of it All   23005139   
7201    5267177                Spacious 1BR Bushwick Surfers Loft!   27263990   

      host_name neighbourhood_group       neighbourhood  latitude  longitude  \
18907     Alexa              Queens            Elmhurst  40.74596  -73.87762   
46663      Nick              Queens             Astoria  40.75883  -73.92160   
19757     Jamie            Brooklyn         Fort Greene  40.69670  -73.97477   
9705       Aude            Brooklyn          Greenpoint  40.72435  -73.95154   
3322      S & G           Manhattan        East Village  40.72754  -73.98555   
12637     Kerry            Brooklyn  Bedford-Stuyvesant  40.68382  -73.95566   
19383    Nicole            Brooklyn       Crown Heights  40.67747  -73.94660   
35358    Daphna              Queens            Woodside  40.74588  -73.89721   
28001  Samantha            Brooklyn       Crown Heights  40.67512  -73.96185   
7201    Lindsey            Brooklyn            Bushwick  40.69544  -73.93046   

             room_type  price  minimum_nights  number_of_reviews last_review  \
18907     Private room     60               2                  0         NaN   
46663  Entire home/apt    250               2                  0         NaN   
19757     Private room     80               1                 53  2018-06-01   
9705      Private room     95               1                119  2019-06-24   
3322   Entire home/apt    450               3                 12  2019-01-02   
12637  Entire home/apt    300               2                 24  2017-07-04   
19383  Entire home/apt     90               9                  7  2019-05-12   
35358     Private room     45               4                  4  2018-12-14   
28001     Private room     55               3                  0         NaN   
7201   Entire home/apt    110               3                  0         NaN   

       reviews_per_month  calculated_host_listings_count  availability_365  
18907                NaN                               2                66  
46663                NaN                               1               282  
19757               2.15                               1                 0  
9705                2.57                               2                 1  
3322                0.19                               2                 0  
12637               0.56                               1                 0  
19383               0.54                               1                 0  
35358               0.43                               1                 0  
28001                NaN                               1                 0  
7201                 NaN                               1                 0

## Finding dimensions, shape, size of the dataframe using Pandas.

In [47]:
# how many rows and columns the dataset has
df.shape

(48895, 16)

In [48]:
#First columns

df.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [49]:
#last columns

df.tail()

id                                               name   host_id  \
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23

In [50]:
#Data types and null values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [51]:
df.isna().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [52]:
df.isnull().values.sum()

20141

### Conclusions

- The airbnb data set has 48895 rows and 16 columns
- It has 10 numerical features and 6 categorical features.
- There is missing data: Feature 'name' has 16 null values. Feature 'host_name' has 21 null values. Features "last_review" and " reviews_per_month" have 10052 null values.
- There is a total of 20141 null values.

## Checking for duplicates

In [53]:
df.duplicated().sum()

0

There are no duplicated rows.

In [54]:
df['host_id'].duplicated().sum()

11438

There seems to be more than one entry per host.

## Unique values


In [55]:
df.nunique()

id                                48895
name                              47905
host_id                           37457
host_name                         11452
neighbourhood_group                   5
neighbourhood                       221
latitude                          19048
longitude                         14718
room_type                             3
price                               674
minimum_nights                      109
number_of_reviews                   394
last_review                        1764
reviews_per_month                   937
calculated_host_listings_count       47
availability_365                    366
dtype: int64

## Descriptive Statistics

In [56]:
# summary of all the numerical variables present in our data frame

round(df.describe(),2)

id       host_id  latitude  longitude     price  \
count     48895.00  4.889500e+04  48895.00   48895.00  48895.00   
mean   19017143.24  6.762001e+07     40.73     -73.95    152.72   
std    10983108.39  7.861097e+07      0.05       0.05    240.15   
min        2539.00  2.438000e+03     40.50     -74.24      0.00   
25%     9471945.00  7.822033e+06     40.69     -73.98     69.00   
50%    19677284.00  3.079382e+07     40.72     -73.96    106.00   
75%    29152178.50  1.074344e+08     40.76     -73.94    175.00   
max    36487245.00  2.743213e+08     40.91     -73.71  10000.00   

       minimum_nights  number_of_reviews  reviews_per_month  \
count        48895.00           48895.00           38843.00   
mean             7.03              23.27               1.37   
std             20.51              44.55               1.68   
min              1.00               0.00               0.01   
25%              1.00               1.00               0.19   
50%              3.00               5.00               0.72   
75%              5.00              24.00               2.02   
max           1250.00             629.00              58.50   

       calculated_host_listings_count  availability_365  
count                        48895.00          48895.00  
mean                             7.14            112.78  
std                             32.95            131.62  
min                              1.00              0.00  
25%                              1.00              0.00  
50%                              1.00             45.00  
75%                              2.00            227.00  
max                            327.00            365.00

In [57]:
# summary of all the categorical variables present in our data frame
df.describe(include = 'object')

name host_name neighbourhood_group neighbourhood  \
count            48879     48874               48895         48895   
unique           47905     11452                   5           221   
top     Hillside Hotel   Michael           Manhattan  Williamsburg   
freq                18       417               21661          3920   

              room_type last_review  
count             48895       38843  
unique                3        1764  
top     Entire home/apt  2019-06-23  
freq              25409        1413

## Visualizations and finding relationships

### What can we learn about different hosts and areas?

### Which hosts are the busiest and why?

### Is there any noticeable difference of traffic among different areas and what could be the reason for it?